### PR TITLE
feat: allow deployment webhook to push deployments to multiple repos …

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
feat: allow deployment webhook to push deployments to multiple repos in one request

[BE] Support the new payload
[BE] Add commit_msg to table cicd_deployment_commits
[FE] Update webhook deployment example

### Does this close any open issues?
Related to #6262 

### Screenshots
img_v3_025k_3ef00b6a-f485-4606-a3f4-764d0bf15d9g
img_v3_025k_77d11025-3c51-488e-8eea-6eb05096449g
img_v3_025k_5376bdbe-e691-4c77-bd99-6d082d788edg

img_v3_025k_8f0a63b2-7602-4f9e-b848-afcde9ce168g
img_v3_025k_4f907ed3-8d7d-4019-9f03-ce3f51159a9g
img_v3_025k_dc9144ae-9903-4502-b2e5-ab1be965463g

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by klesh — APPROVED** [review]

File: backend/plugins/webhook/api/deployments.go [hunk 1]
```diff
@@ -84,60 +98,123 @@ func PostDeploymentCicdTask(input *plugin.ApiResourceInput) (*plugin.ApiResource
...
-		RepoId:           request.RepoId,
-		RepoUrl:          request.RepoUrl,
+	if request.Environment == "" {
+		request.Environment = devops.PRODUCTION
 	}
-	err = tx.CreateOrUpdate(deploymentCommit)
-	if err != nil {
-		logger.Error(err, "create deployment commit")
-		return nil, err
+	duration := request.FinishedDate.Sub(*request.StartedDate).Seconds()
```
Comment (d4x1) [inline backend/plugins/webhook/api/deployments.go:117]: `duration` is float type now. So please use `.Milliseconds()`, and divide 1000, get the float64 duration.

File: backend/plugins/webhook/api/deployments.go [hunk 2]
```diff
@@ -84,60 +98,123 @@ func PostDeploymentCicdTask(input *plugin.ApiResourceInput) (*plugin.ApiResource
...
+	name := request.Name
+	if name == "" {
+		if request.DeploymentCommits == nil {
+			name = fmt.Sprintf(`deployment for %s`, request.CommitSha)
+		} else {
+			commit_sha_list := []string{}
```
Comment (d4x1) [inline backend/plugins/webhook/api/deployments.go:123]: why not name it with `commitShaList`?

File: backend/plugins/webhook/api/deployments.go [hunk 3]
```diff
@@ -84,60 +98,123 @@ func PostDeploymentCicdTask(input *plugin.ApiResourceInput) (*plugin.ApiResource
...
+			for _, commit := range request.DeploymentCommits {
+				commit_sha_list = append(commit_sha_list, commit.CommitSha)
```
Comment (d4x1) [inline backend/plugins/webhook/api/deployments.go:125]: same as above.

File: backend/plugins/webhook/api/deployments.go [hunk 4]
```diff
@@ -84,60 +98,123 @@ func PostDeploymentCicdTask(input *plugin.ApiResourceInput) (*plugin.ApiResource
...
+		}
+
+		// create a deployment record
+		if err = tx.CreateOrUpdate(deploymentCommit.ToDeployment()); err != nil {
+			logger.Error(err, "create deployment")
+			return nil, err
+		}
+	} else {
+		for _, commit := range request.DeploymentCommits {
+			urlHash16 := fmt.Sprintf("%x", md5.Sum([]byte(commit.RepoUrl)))[:16]
```
Comment (d4x1) [inline backend/plugins/webhook/api/deployments.go:176]: I think it is duplicated code, and can be wrapped to a function.

File: backend/plugins/webhook/api/deployments.go [hunk 5]
```diff
@@ -36,18 +38,30 @@ import (
...
+	RepoId string `mapstructure:"repo_id"`
+	Result string `mapstructure:"result"`
 	// start_time and end_time is more readable for users,
 	// StartedDate and FinishedDate is same as columns in db.
 	// So they all keep.
 	CreatedDate  *time.Time `mapstructure:"create_time"`
 	StartedDate  *time.Time `mapstructure:"start_time" validate:"required"`
 	FinishedDate *time.Time `mapstructure:"end_time"`
+	RepoUrl      string     `mapstructure:"repo_url"`
 	Environment  string     `validate:"omitempty,oneof=PRODUCTION STAGING TESTING DEVELOPMENT"`
```
Comment (d4x1) [inline backend/plugins/webhook/api/deployments.go:50]: why doesn't  this field have `mapstructure`?

File: backend/plugins/webhook/api/deployments.go (hunk 4 above)
Comment (abeizn) [reply backend/plugins/webhook/api/deployments.go:176]: no, this is in the loop

File: backend/plugins/webhook/api/deployments.go (hunk 5 above)
Comment (abeizn) [reply backend/plugins/webhook/api/deployments.go:50]: This will not change the name, it will be resolved into **_environment_**

**Comment by abeizn** [comment]
all fixed by other pr

**Pull request merged**